.github/workflows/ci.yml

```diff
@@ -35,21 +35,52 @@ jobs:
         run: |
           sudo add-apt-repository multiverse
           sudo apt update
-          sudo apt install rar unrar
+          sudo apt install rar unrar libssl-dev libcurl4-openssl-dev zlib1g-dev
       - name: Setup Python ${{ matrix.python-version }}
         uses: actions/setup-python@v2
         with:
           python-version: ${{ matrix.python-version }}
+      - name: Setup msbuild on Windows
+        if: matrix.os == 'windows-latest'
+        uses: microsoft/setup-msbuild@v1.1
+      - name: Set up Visual Studio shell
+        if: matrix.os == 'windows-latest'
+        uses: egor-tensin/vs-shell@v2
+        with:
+          arch: x64
       - name: Check out source repository
         uses: actions/checkout@v2
       - name: Install dependencies
         run: |
           pip3 install -r requirements.txt
           pip3 install --pre torch -f https://download.pytorch.org/whl/nightly/cpu/torch_nightly.html
+          pip3 install cmake ninja pybind11 pybind11-global
+          echo "/home/runner/.local/bin" >> $GITHUB_PATH
       - name: Install test requirements
         run: pip3 install expecttest fsspec iopath==0.1.9 numpy pytest rarfile
+      - name: Install AWS-SDK-CPP on Windows for S3 IO datapipes
+        if: matrix.os == 'windows-latest'
+        run: |
+          git clone --recurse-submodules https://github.com/aws/aws-sdk-cpp
+          cd aws-sdk-cpp
+          cmake -S . -B build -GNinja -DBUILD_ONLY="s3;transfer" -DENABLE_TESTING=OFF -DCMAKE_BUILD_TYPE=Release
+          cmake --build build --config Release
+          cmake --install build --config Release
+      - name: Install AWS-SDK-CPP on Non-Windows for S3 IO datapipes
+        if: matrix.os != 'windows-latest'
+        run: |
+          git clone --recurse-submodules https://github.com/aws/aws-sdk-cpp
+          cd aws-sdk-cpp/
+          mkdir sdk-build
+          cd sdk-build
+          cmake .. -DCMAKE_BUILD_TYPE=Release -DBUILD_ONLY="s3;transfer" -DAUTORUN_UNIT_TESTS=OFF
+          make
+          sudo make install
       - name: Build TorchData
         run: python setup.py develop
+        env:
+          BUILD_S3: 1
+          BUILD_PYTHON_VERSION: ${{ matrix.python-version }}
```

Comment:
Could we add test when `BUILD_S3` was disabled?

Reply:
I think I've added a test for missing `_torchdata_` as well as for missing `aws-sdk-cpp`.

To run those tests, should we build two binaries? and also have two separate test stages?

Reply:
What you can do is to add another matrix option here https://github.com/pytorch/data/blob/19cf4530084820c54264141f22b82ba0e2997cfd/.github/workflows/ci.yml#L23-L31
```
with-s3:
  - 1
  - 0
```
Then, using this option to determine if we want to build with S3.
```
env:
    BUILD_S3: ${{ matrix.with-s3 }}
```

Reply:
updated the CI for a new matrix variable for BUILD_S3, and related tests have passed.